core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -565,7 +565,7 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
           InodeDirectory.create(mDirectoryIdGenerator.getNewDirectoryId(),
               currentInodeDirectory.getId(), pathComponents[k], missingDirOptions);
       // Lock the newly created inode before subsequent operations, and add it to the lock group.
-      lockList.lockWrite(dir);
+      lockList.lockWriteAndCheckFullPath(dir, currentInodeDirectory, pathComponents[k]);
```

Comment:
- Is it possible to see `InvalidPathException` here due to concurrent rename? shall we retry if `InvalidPathException` is thrown? It seems now the error is directly thrown back to user.  
- More generally, in some other methods, the same exception is caught and continue. Do we have a guideline when we ignore errors and when not?